### DISCUSSION TITLE
Variable mass

### DISCUSSION THREAD

**Discussion by agodemar** [JSBSim-Team/jsbsim#390] [Ideas]
It would be good to have a point mass with a variable value.
For instance, this would model a tank that would leak fuel. Or a tank that would be filled during a refueling manoeuvre. 
This mass variation in time should be taken into account at each time step when aircraft mass & balance are recalculated.

**Comment by seanmcleod** [comment]
So you can from the outside periodically update the point mass's weight. Which will also update it's moment of inertia etc. based on the shape type.

```c++
// This function binds properties for each pointmass object created.
//
void FGMassBalance::PointMass::bind(FGPropertyManager* PropertyManager,
                                    unsigned int num) {
  string tmp = CreateIndexedPropertyName("inertia/pointmass-weight-lbs", num);
  PropertyManager->Tie( tmp.c_str(), this, &PointMass::GetPointMassWeight,
                                       &PointMass::SetPointMassWeight);
```

That's for a point mass.

Now if you're specifically talking about fuel being added or lost then you don't want to use a point mass, you want to deal with an actual fuel tank.

```c++
  property_name = base_property_name + "/external-flow-rate-pps";
  PropertyManager->Tie( property_name.c_str(), (FGTank*)this, &FGTank::GetExternalFlow,
                                       &FGTank::SetExternalFlow );
```

```c++
double FGTank::Calculate(double dt, double TAT_C)
{
  if(ExternalFlow < 0.) Drain( -ExternalFlow *dt);
  else Fill(ExternalFlow * dt);
```

**Reply by bcoconni** [reply]
Just for the sake of completeness: the features mentioned by @seanmcleod are ignoring the impulse component `v*dm/dt`.  However in cases where this term is significant (i.e. meanly for rockets), it is simulated by propulsion elements.

**Comment by agodemar** [comment]
@seanmcleod Can a point mass be set to a variable mass, and the variable value be changed in a script according to a user-defined function?

**Comment by seanmcleod** [comment]
The PointMass object doesn't have an equivalent `mass-rate-pps` property equivalent to the fuel tank's `external-flow-rate-pps` property. So if you want the point mass's mass to be variable then your script needs to periodically update it's mass/weight by writing to it's `inertia/pointmass-weight-lbs` property. 

If you're trying to model fuel in terms of leaks or getting additional fuel via aerial re-fueling etc. I'd suggest using the fuel tank object and not a point mass.
